typescript/cli/src/deploy/warp.ts

```diff
@@ -206,29 +231,118 @@ async function runDeployPlanStep({ context, warpDeployConfig }: DeployParams) {
 async function executeDeploy(
   params: DeployParams,
   apiKeys: ChainMap<string>,
-): Promise<HyperlaneContractsMap<HypERC20Factories | HypERC721Factories>> {
+): Promise<{
+  deployedContracts: ChainMap<Address>;
+  deployments: WarpCoreConfig;
+}> {
   logBlue('🚀 All systems ready, captain! Beginning deployment...');
 
   const {
     warpDeployConfig,
-    context: { multiProvider, isDryRun, dryRunChain, registry },
+    context: { multiProvider, isDryRun, dryRunChain, multiProtocolSigner },
   } = params;
 
+  assert(multiProtocolSigner, `multiProtocolSigner not defined`);
+
   const config: WarpRouteDeployConfigMailboxRequired =
     isDryRun && dryRunChain
       ? { [dryRunChain]: warpDeployConfig[dryRunChain] }
       : warpDeployConfig;
-  const registryAddresses = await registry.getAddresses();
 
-  const deployedContracts = await executeWarpDeploy(
+  const contractVerifier = new ContractVerifier(
     multiProvider,
-    config,
-    registryAddresses,
     apiKeys,
+    coreBuildArtifact,
+    ExplorerLicenseType.MIT,
+  );
+
+  const ismFactoryDeployer = new HyperlaneProxyFactoryDeployer(
+    multiProvider,
+    contractVerifier,
+  );
+
+  // For each chain in WarpRouteConfig, deploy each Ism Factory, if it's not in the registry
+  // Then return a modified config with the ism and/or hook address as a string
+  const modifiedConfig = await resolveWarpIsmAndHook(
+    config,
+    params.context,
+    ismFactoryDeployer,
+    contractVerifier,
   );
 
+  let deployedContracts: ChainMap<Address> = {};
+  const deployments: WarpCoreConfig = { tokens: [] };
+
+  // get unique list of protocols
+  const protocols = Array.from(
+    new Set(
+      Object.keys(modifiedConfig).map((chainName) =>
+        multiProvider.getProtocol(chainName),
+      ),
+    ),
+  );
+
+  for (const protocol of protocols) {
+    const protocolSpecificConfig = objFilter(
+      modifiedConfig,
+      (chainName, _): _ is any =>
+        multiProvider.getProtocol(chainName) === protocol,
+    );
+
+    if (isObjEmpty(protocolSpecificConfig)) {
+      continue;
+    }
+
+    switch (protocol) {
```

Comment:
At first glance, this is a significant revamp of the warp route deployment logic that lived in `typescript/sdk/src/token`. How are we verifying that the logic (i.e. the set of transactions generated by a deployment run) stays the same through this refactoring?